odilia/src/tower.rs

```diff
@@ -151,20 +383,21 @@ where
 	fn call(&mut self, req: I) -> Self::Future {
 		let len = self.inner.len();
 		let ic = self.inner.clone();
-		SerialServiceFuture { inner: ic, req, results: Vec::with_capacity(len) }
+		let inner = std::mem::replace(&mut self.inner, ic);
+		SerialServiceFuture { inner, req, results: Vec::with_capacity(len) }
 	}
 }
 
 pub struct Handlers<S> {
 	state: S,
 	atspi_handlers:
 		HashMap<(&'static str, &'static str), Vec<BoxService<Event, Response, Error>>>,
-	command_handlers: BTreeMap<CommandDiscriminants, Vec<BoxService<Command, (), Error>>>,
+	command_handlers: BTreeMap<CommandDiscriminants, BoxService<Command, (), Error>>,
```

Comment:
It's a way to disambiguate between different variants. Since I'm adding them to a global handler variable, this is the best way to differentiate them.